Add POI option

thomasgubler opened the pull request:
The initial list of POI I added in data.js can of course be extended in the future based on https://wiki.openstreetmap.org/wiki/Map_features . The current choice is based on my personal use.

Initial Marker colors:
![Screenshot from 2022-02-15 23-09-25](https://user-images.githubusercontent.com/1419479/154157941-e5e8e9df-b526-42e7-9391-ed9055eac72b.png)

Updated marker colors after review feedback:
![Screenshot from 2022-03-12 11-55-54](https://user-images.githubusercontent.com/1419479/158015283-5e2f83f5-1447-434a-8a1a-da9f3329ca47.png)

thomasgubler commented:
@vcoppe do you have any feedback for this pull request? thank you.

vcoppe commented:
Thanks for your work with this! It is a very nice feature.
I only have 2 remarks/suggestions:
- we could probably have a single L.OverpassLayer making a single request to Overpass with all the POI types selected, it would reduce a lot the number of requests (for now I am getting a lot of "Too many requests" errors) which the least we can do for this free service
- we should maybe find another POI icon as the one used for waypoints to avoid confusion
I can try do these changes myself if you want

thomasgubler commented:
Thanks for reviewing.

**Update:**

- changed marker color (see screenshot in description)
- investigated "too many requests" error from overpass API
  - OverpassLayer does not support bundling all layers in all requests, also we would need to deal then with separating the different POI types for display after getting the request ourselves (right now this is handled by multiple OverpassLayers)
  - I found that also with single requests for 1 POI layer I got "too many requests" or similar already on first request or when moving the map once
  - switched to the mirror from komi.systems (documented here https://wiki.openstreetmap.org/wiki/Overpass_API) which seems robust against reasonable requests

**Todo:**
- layers loading got refactored on master meanwhile --> need to update to new structure

thomasgubler commented:
updated to map layer refactoring on master via merge from master and manual conflict resolving

vcoppe commented:
Perfect! It does indeed seem much more responsive with the new Overpass endpoint.
Thanks for your ideas and the implementation!
